src/Microsoft.IdentityModel.Tokens/SignatureProvider.cs

```diff
@@ -156,5 +156,7 @@ public virtual bool Verify(byte[] input, int inputOffset, int inputLength, byte[
         /// Gets or sets a bool indicating if this <see cref="SignatureProvider"/> is expected to create signatures.
         /// </summary>
         public bool WillCreateSignatures { get; protected set; }
+
+        internal bool WasCached { get; set; }
```

Comment:
nit: `IsCached`? I think implies the current state a bit clearer.

Reply:
@pmaytak I agree, will change.